app/src/main/java/com/weatherxm/data/datasource/QuestsDataSource.kt

```diff
@@ -23,6 +23,10 @@ interface QuestsDataSource {
     fun fetchOnboardingProgress(userId: String): Either<Throwable, QuestUserProgress>
     suspend fun fetchOnboardingQuest(): Either<Throwable, QuestWithStepsFirestore>
     suspend fun completeQuest(userId: String, questId: String): Either<Throwable, Unit>
+    fun markOnboardingStepAsCompleted(userId: String, stepId: String)
+    fun markOnboardingStepAsSkipped(userId: String, stepId: String)
+    fun removeOnboardingStepFromCompleted(userId: String, stepId: String)
+    fun removeOnboardingStepFromSkipped(userId: String, stepId: String)
```

Comment:
Maybe let's make a minor change here, to include `questId` as we do in `completeQuest` so to make them reusable? I know we said it's not needed atm, but it's a very easy fix.

So they would be:
    fun markQuestStepAsCompleted(userId: String, questId: String, stepId: String)
    fun markQuestStepAsSkipped(userId: String, questId: String, stepId: String)
    fun removeQuestStepFromCompleted(userId: String, questId: String, stepId: String)
    fun removeQuestStepFromSkipped(userId: String, questId: String, stepId: String)